guava/src/com/google/common/math/IntMath.java

```diff
@@ -99,8 +105,10 @@ public static boolean isPowerOfTwo(int x) {
    * a signed int. The implementation is branch-free, and benchmarks suggest it is measurably (if
    * narrowly) faster than the straightforward ternary expression.
    */
+  @SuppressWarnings("return.type.incompatible")// Since `Integer.SIZE - 1` bits form is: 1111, for shifted( >>> ) positive values return 1,
+  //otherwise return 0.
```

Comment:
This is an excellent explanation. It's a good example of the point I was trying to make above (probably better than the one I gave), because it explains why this is safe in *this* code.

Reply:
That said the suppress warning key should name the checker that issues the warning.

Reply:
@kelloggm I've tried both `lowerbound` and `upperbound` as prefixes but it would not work. May I ask what is the appropriate prefix for this suppress warning.

Reply:
`value`, because `IntRange` is an annotation from the value checker. You can see this from the import of the annotation: it comes from `common.value.qual` instead of `checker.index.qual` - the last name before "qual" is the suppress warning key, by convention.